client/dashboard/customizable.js

```diff
@@ -163,29 +163,34 @@ class CustomizableDashboard extends Component {
 	render() {
 		const { query, path } = this.props;
 		const { sections } = this.state;
-		const visibleSections = sections.filter( section => section.isVisible );
+		const visibleSectionKeys = sections
+			.filter( section => section.isVisible )
+			.map( section => section.key );
 
 		return (
 			<Fragment>
 				<H>{ __( 'Customizable Dashboard', 'woocommerce-admin' ) }</H>
 				<ReportFilters query={ query } path={ path } />
-				{ visibleSections.map( ( section, index ) => {
-					return (
-						<Section
-							component={ section.component }
-							hiddenBlocks={ section.hiddenBlocks }
-							key={ section.key }
-							onChangeHiddenBlocks={ this.onChangeHiddenBlocks( section.key ) }
-							onTitleUpdate={ this.onSectionTitleUpdate( section.key ) }
-							path={ path }
-							query={ query }
-							title={ section.title }
-							onMove={ partial( this.onMove, index ) }
-							onRemove={ this.toggleVisibility( section.key ) }
-							isFirst={ 0 === index }
-							isLast={ visibleSections.length === index + 1 }
-						/>
-					);
+				{ sections.map( ( section, index ) => {
+					if ( section.isVisible ) {
+						return (
+							<Section
+								component={ section.component }
+								hiddenBlocks={ section.hiddenBlocks }
+								key={ section.key }
+								onChangeHiddenBlocks={ this.onChangeHiddenBlocks( section.key ) }
+								onTitleUpdate={ this.onSectionTitleUpdate( section.key ) }
+								path={ path }
+								query={ query }
+								title={ section.title }
+								onMove={ partial( this.onMove, index ) }
+								onRemove={ this.toggleVisibility( section.key ) }
+								isFirst={ section.key === visibleSectionKeys[ 0 ] }
```

Comment:
Somewhat unrelated, but might be a good follow-up here: I wonder if passing `visibleSectionKeys` down instead of `isFirst` and `isLast` props would prove useful to plugins that plan to use this section more dynamically.  This would also have the added benefit of reducing the number of props we need to pass down.

Reply:
Yeah good idea, that would give more control to extensions. But then we'd require them to apply the logic and render Controls. I'll give this a bit more thought.

```js
<Controls
	onToggle={ onToggle }
	onMove={ onMove }
	onRemove={ onRemove }
	isFirst={ 'leaderboards' === visibleSectionKeys[ 0 ] }
        isLast={ 'leaderboards' === visibleSectionKeys[ visibleSectionKeys.length - 1 ] }
/>
```

Reply:
Ya, that's true.  Could we pass `visibleSectionKeys` and the key down to the control and handle logic there?  That way they're only passing props and no logic changes are necessary inside extensions.

Reply:
Yeah, this is the ideal scenario. I'll merge this in and give some thought on how to make this happen.